WalletWasabi/Extensions/NBitcoinExtensions.cs

```diff
@@ -150,9 +150,10 @@ public static int GetAnonymitySet(this Transaction me, uint outputIndex)
 		return anonsets;
 	}
 
-	public static bool IsLikelyCoinjoin(this Transaction me)
-		=> me.Inputs.Count > 1 // The tx must have more than one input in order to be a coinjoin.
-		&& me.HasIndistinguishableOutputs(); // The tx must have more than one equal output in order to be a coinjoin.
+	public static bool IsLikelyCoinjoin(this SmartTransaction me)
+		=> me.Transaction.Inputs.Count > 1 // The tx must have more than one input in order to be a coinjoin.
+		&& me.Transaction.Inputs.Count != me.WalletInputs.Count
+		&& me.Transaction.HasIndistinguishableOutputs(); // The tx must have more than one equal output in order to be a coinjoin.
```

Comment:
A coinjoin is a tx in witch:
* I am not the only one providing inputs 
* I am not the only one receiving money (otherwise it is a payjoin payment to my wallet)
* There are indistinguishable outputs (equal amounts with equal scriptPubKey type)

Reply:
How about this:
```
public static bool IsLikelyCoinjoin(this SmartTransaction me)
	=> me.Transaction.Inputs.Count > 1 // The tx must have more than one input in order to be a coinjoin.
	&& me.Transaction.Inputs.Count != me.WalletInputs.Count // The tx must have overall more inputs then what one own to be a coinjoin.
	&& me.Transaction.Outputs.Count != me.WalletOutputs.Count // The tx mush have overall more outputs then what one own to be a coinjoin.
	&& me.Transaction.HasIndistinguishableOutputs(); // The tx must have more than one equal output in order to be a coinjoin.
```

Reply:
`me.Transaction.Inputs.Count > 1` is redundant. `HasIndistinguishableOutputs` doesn't take scriptPubKeyType into account (we can ignore it if we don't care about taproot scripts but I think we should prefer correctness)

Reply:
> I am not the only one receiving money (otherwise it is a payjoin payment to my wallet)

PayJoin is also a coinjoin, but let's not let such minor details ruin functionality. Added.

> me.Transaction.Inputs.Count > 1

Removed.

> doesn't take scriptPubKeyType into account

AFAIK enforcing equal scriptPubKey types is like enforcing scriptPubKeys those start with the same letter 😄